### PR TITLE
TEST-#2288: Cover by tests delimiters parameters of read_csv

### LINKED ISSUE #2288: Cover by tests delimiters parameters of read_csv
According to source files (https://github.com/pandas-dev/pandas/blob/v1.1.3/pandas/io/parsers.py#L540-L597) `pandas.read_csv` parameters roughly divided into several groups, this issue aims to cover `delimiter` parameters.

### PR DESCRIPTION
**Pull request by amyskov**
Signed-off-by: Alexander Myskov <alexander.myskov@intel.com>

## What do these changes do?

- [ ] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/developer/contributing.html)
- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #2288  
- [ ] tests added and passing

### CODE REVIEW

File: modin/pandas/test/test_io.py [hunk 1]
```diff
@@ -60,8 +62,21 @@
 TEST_GBQ_FILENAME = "test_gbq."
 SMALL_ROW_SIZE = 2000
 
+test_data_dir = os.path.join(os.path.dirname(__file__), "data")
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:65]: The name should let us know that this folder can be deleted.

File: modin/pandas/test/test_io.py [hunk 2]
```diff
@@ -173,48 +189,191 @@ def teardown_test_file(test_path):
         os.remove(test_path)
 
 
-@pytest.fixture
-def make_csv_file(delimiter=",", compression="infer"):
-    """Pytest fixture factory that makes temp csv files for testing.
+def get_random_string():
+    random_string = "".join(
+        random_state.choice(
+            [x for x in ascii_uppercase + ascii_lowercase + digits], size=10
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:195]: Why is it so hard to create unique strings?

I mean, can't we get by with a smaller set of characters from which strings will be created?

File: modin/pandas/test/test_io.py [hunk 3]
```diff
@@ -60,8 +62,21 @@
...
+
+def eval_io(
+    fn_name,
+    comparator=df_equals,
+    cast_to_str=False,
+    check_exception_type=True,
+    nonacceptable_exception_types=read_csv_non_acc_exc,
+    *args,
+    **kwargs,
+):
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:79]: Please add docstring.

File: modin/pandas/test/test_io.py [hunk 4]
```diff
@@ -173,48 +189,191 @@ def teardown_test_file(test_path):
         os.remove(test_path)
 
 
-@pytest.fixture
-def make_csv_file(delimiter=",", compression="infer"):
-    """Pytest fixture factory that makes temp csv files for testing.
+def get_random_string():
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:192]: Request docstring.

File: modin/pandas/test/test_io.py [hunk 5]
```diff
@@ -60,8 +62,21 @@
 TEST_GBQ_FILENAME = "test_gbq."
 SMALL_ROW_SIZE = 2000
 
+test_data_dir = os.path.join(os.path.dirname(__file__), "data")
 
-def eval_io(path, fn_name, comparator=df_equals, cast_to_str=False, *args, **kwargs):
+read_csv_non_acc_exc = [TypeError, FileNotFoundError]
+read_csv_non_acc_exc_without_TypeError = list(read_csv_non_acc_exc).remove(TypeError)
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:68]: Complex names.

File: modin/pandas/test/test_io.py [hunk 6]
```diff
@@ -173,48 +189,191 @@ def teardown_test_file(test_path):
...
+def get_random_string():
+    random_string = "".join(
+        random_state.choice(
+            [x for x in ascii_uppercase + ascii_lowercase + digits], size=10
+        ).tolist()
+    )
+    return random_string
+
+
+def insert_lines_to_csv(
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:201]: Request docstring.

File: modin/pandas/test/test_io.py [hunk 7]
```diff
@@ -173,48 +189,191 @@ def teardown_test_file(test_path):
...
-            df = pandas.DataFrame(
-                {
-                    "col1": np.arange(row_size),
-                    "col2": [str(x.date()) for x in dates],
-                    "col3": np.arange(row_size),
-                    "col4": [str(x.time()) for x in dates],
-                }
-            )
+            data = {
+                "col1": np.arange(row_size) * 10,
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:277]: Doesn't increase columns size by this way. Instead increase row_size directly. 

Defining from one constant makes it easier to maintain.

File: modin/pandas/test/test_io.py [hunk 8]
```diff
@@ -173,48 +189,191 @@ def teardown_test_file(test_path):
...
+            if thousands_separator:
+                df["col1"] = df["col1"].apply(
+                    lambda x: f"{x:,d}".replace(",", thousands_separator)
+                )
+                df["col3"] = df["col3"].apply(
+                    lambda x: f"{x:,d}".replace(",", thousands_separator)
+                )
+                df["col6"] = df["col6"].apply(
+                    lambda x: f"{x:,f}".replace(",", thousands_separator)
+                )
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:315]: This can be done in a loop.

File: modin/pandas/test/test_io.py [hunk 9]
```diff
@@ -423,6 +582,68 @@ def teardown_fwf_file():
             pass
 
 
+def get_unique_filename(
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:585]: Request docstring.

**Review by anmyachev — CHANGES_REQUESTED** [review]

File: modin/pandas/test/test_io.py (hunk 1 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py:65]: This is common directory for test data that is already used by other tests. Do you mean we need to create separate directory for `read_csv` tests?

File: modin/pandas/test/test_io.py (hunk 2 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py:216]: I'm not sure if I understood you. We can create test string from smaller set of characters, but why we should do?

File: modin/pandas/test/test_io.py (hunk 3 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py]: Docstring is added.

File: modin/pandas/test/test_io.py (hunk 4 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py:213]: Docstring is added.

File: modin/pandas/test/test_io.py (hunk 5 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py]: Names are simplified.

File: modin/pandas/test/test_io.py (hunk 6 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py:222]: Docstring is added.

File: modin/pandas/test/test_io.py (hunk 7 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py:312]: Here increased only values, not a number of elements. This is done to test `thousands` parameter.

File: modin/pandas/test/test_io.py (hunk 8 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py]: Ok, loop is added.

File: modin/pandas/test/test_io.py (hunk 9 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py:618]: Docstring is added.

File: modin/pandas/test/test_io.py (hunk 1 above)
Comment (anmyachev) [reply modin/pandas/test/test_io.py:65]: Yes, do not mix the generated data with source data from github.

File: modin/pandas/test/utils.py [hunk 10]
```diff
@@ -630,8 +630,14 @@ def eval_general(
     comparator=df_equals,
     __inplace__=False,
     check_exception_type=True,
+    nonacceptable_exception_types=None,
+    check_kwargs_callable=True,
```
Comment (anmyachev) [inline modin/pandas/test/utils.py:634]: It's unused?

File: modin/pandas/test/test_io.py (hunk 2 above)
Comment (anmyachev) [reply modin/pandas/test/test_io.py:216]: This is Occam's blade. Doing what can be easier is easier.

For example: use only string.ascii_letters.

File: modin/pandas/test/test_io.py (hunk 1 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py]: Ok, separate directory for `read_csv` tests data is used now.

File: modin/pandas/test/test_io.py (hunk 2 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py]: Ok, only `string.ascii_letters` is used now.

File: modin/pandas/test/utils.py (hunk 10 above)
Comment (amyskov) [reply modin/pandas/test/utils.py]: Yes for now (`check_kwargs_callable` parameter is removed).

File: modin/pandas/test/test_io.py (hunk 7 above)
Comment (anmyachev) [reply modin/pandas/test/test_io.py:312]: Oh, I see.

File: modin/pandas/test/utils.py [hunk 11]
```diff
@@ -630,8 +630,13 @@ def eval_general(
     comparator=df_equals,
     __inplace__=False,
     check_exception_type=True,
+    nonacceptable_exception_types=None,
     **kwargs,
 ):
+    if nonacceptable_exception_types not in (None, False):
```
Comment (anmyachev) [inline modin/pandas/test/utils.py:636]: What types can this parameter be?

Can we not use bool values?

File: modin/pandas/test/utils.py [hunk 12]
```diff
@@ -645,6 +650,16 @@ def execute_callable(fn, inplace=False, md_kwargs={}, pd_kwargs={}):
...
+                if check_exception_type is not None and not isinstance(
+                    check_exception_type, bool
+                ):
+                    assert isinstance(
+                        md_e.value, check_exception_type
+                    ), "exception type not in the list of acceptable exception types"
+                if nonacceptable_exception_types is not None:
+                    assert not isinstance(
+                        md_e.value, tuple(nonacceptable_exception_types)
+                    ), "not acceptable exception type"
```
Comment (anmyachev) [inline modin/pandas/test/utils.py:662]: Messages should contain the values of the variables from which the exception occurred. (to simplify debugging)

File: modin/pandas/test/test_io.py [hunk 13]
```diff
@@ -60,13 +62,49 @@
...
+    nonacceptable_exception_types: Exception or list of Exceptions
+        Exceptions types that are prohibited.
+    """
 
-def eval_io(path, fn_name, comparator=df_equals, cast_to_str=False, *args, **kwargs):
     def applyier(module, *args, **kwargs):
         result = getattr(module, fn_name)(*args, **kwargs)
-        # There could be some missmatches in dtypes, so we're
-        # casting the whole frame to `str` before comparison.
-        # See issue #1931 for details.
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:69]: This issue is fixed now?

File: modin/pandas/test/test_io.py [hunk 14]
```diff
@@ -60,13 +62,49 @@
...
+
+def eval_io(
+    fn_name,
+    comparator=df_equals,
+    cast_to_str=False,
+    check_exception_type=True,
+    nonacceptable_exception_types=io_ops_bad_exc,
+    *args,
+    **kwargs,
+):
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:83]: Move the utility functions to the appropriate file.

File: modin/pandas/test/utils.py (hunk 11 above)
Comment (amyskov) [reply modin/pandas/test/utils.py]: `nonacceptable_exception_types ` can be `Exception` or list of `Exceptions`.
Ok, `not nonacceptable_exception_types` statement is used.

File: modin/pandas/test/utils.py (hunk 12 above)
Comment (amyskov) [reply modin/pandas/test/utils.py]: Corrected.

File: modin/pandas/test/test_io.py (hunk 13 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py:69]: No, this description was replaced to the docstring.

File: modin/pandas/test/test_io.py (hunk 14 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py]: Done.

File: modin/pandas/test/test_io.py [hunk 15]
```diff
@@ -423,6 +562,87 @@ def teardown_fwf_file():
...
+
+    Returns
+    -------
+        Unique file name.
+    """
+    # shortcut if kwargs parameter os not provided
+    if len(kwargs) == 0 and extension == "csv" and suffix == "":
+        return os.path.join(data_dir, (test_name + f"_{suffix}" + f".{extension}"))
+
+    assert "." not in extension, "please provide pure extenxion name without '.'"
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:595]:
```suggestion
    assert "." not in extension, "please provide pure extension name without '.'"
```

File: modin/pandas/test/test_io.py [hunk 16]
```diff
@@ -60,25 +62,9 @@
...
-
-    eval_general(
-        pd,
-        pandas,
-        applyier,
-        path=path,
-        *args,
-        **kwargs,
-    )
+test_data_dir = os.path.join(os.path.dirname(__file__), "read_csv_data")
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:65]: Variable in global context should be in upper case.

File: modin/pandas/test/test_io.py [hunk 17]
```diff
@@ -173,48 +159,201 @@ def teardown_test_file(test_path):
         os.remove(test_path)
 
 
-@pytest.fixture
-def make_csv_file(delimiter=",", compression="infer"):
-    """Pytest fixture factory that makes temp csv files for testing.
-
-    Yields:
-        Function that generates csv files
+def get_random_string():
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:162]: This is also a utility.

File: modin/pandas/test/test_io.py [hunk 18]
```diff
@@ -173,48 +159,201 @@ def teardown_test_file(test_path):
...
+    random_string = "".join(
+        random_state.choice([x for x in ascii_letters], size=10).tolist()
+    )
+    return random_string
+
+
+def insert_lines_to_csv(
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:169]: A utility.

File: modin/pandas/test/test_io.py (hunk 16 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py]: Corrected.

File: modin/pandas/test/test_io.py (hunk 17 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py]: Replaced to the `utils.py`.

File: modin/pandas/test/test_io.py (hunk 18 above)
Comment (amyskov) [reply modin/pandas/test/test_io.py]: Replaced to the `utils.py`.

**Review by anmyachev — APPROVED** [review]

**Pull request merged**